src/ring_theory/power_basis.lean

```diff
@@ -121,6 +121,15 @@ lemma exists_eq_aeval [nontrivial S] (pb : power_basis R S) (y : S) :
   ∃ f : polynomial R, f.nat_degree < pb.dim ∧ y = aeval pb.gen f :=
 (mem_span_pow pb.dim_ne_zero).mp (by simpa using pb.basis.mem_span y)
 
+lemma alg_hom_ext [nontrivial S] {S' : Type*} [semiring S'] [algebra R S']
+  (pb : power_basis R S) ⦃f g : S →ₐ[R] S'⦄ (h : f pb.gen = g pb.gen) :
+  f = g :=
+begin
+  ext x,
+  obtain ⟨f, hf, rfl⟩ := pb.exists_eq_aeval x,
+  rw [← polynomial.aeval_alg_hom_apply, ← polynomial.aeval_alg_hom_apply, h]
+end
```

Comment:
No need for `nontrivial S` here. If you add this to `algebra/algebra/basic`:
```lean
-- TODO[gh-6025]: make this an instance once safe to do so
def alg_hom.subsingleton {R A B} [comm_semiring R] [semiring A] [semiring B]
  [algebra R A] [algebra R B] [subsingleton A] :
  subsingleton (A →ₐ[R] B) :=
⟨λ f g, alg_hom.ext $ λ x, subsingleton.elim 0 x ▸ f.map_zero.trans g.map_zero.symm ⟩
```
then you can prove this with
```suggestion
-- TODO[gh-6025]: make this globally an instance once safe to do so
local attribute [instance] alg_hom.subsingleton

lemma alg_hom_ext {S' : Type*} [semiring S'] [algebra R S']
  (pb : power_basis R S) ⦃f g : S →ₐ[R] S'⦄ (h : f pb.gen = g pb.gen) :
  f = g :=
begin
  nontriviality S,
  ext x,
  obtain ⟨f, hf, rfl⟩ := pb.exists_eq_aeval x,
  rw [← polynomial.aeval_alg_hom_apply, ← polynomial.aeval_alg_hom_apply, h]
end
```